src/taipy/core/scenario/scenario.py

```diff
@@ -134,20 +134,27 @@ def pipelines(self, pipelines: Union[List[PipelineId], List[Pipeline]]):
         self._pipelines = pipelines
 
     @property
-    def tasks(self) -> Dict[str, Task]:
-        task_manager = _TaskManagerFactory._build_manager()
-        tasks = {}
-        list_tasks = [pipeline.tasks for pipeline in self.pipelines.values()]
-        for task in list_tasks:
-            for k, v in task.items():
-                t = task_manager._get(v, v)
-                if not isinstance(t, Task):
-                    raise NonExistingTask(v)
-                tasks[k] = v
+    def tasks(self) -> Dict[str, List[Task]]:
+        return self._get_tasks()
+
+    def _get_tasks(self) -> Dict[str, List[Task]]:
```

Comment:
sure that's good with me too 🗡️

Reply:
to answer your point below "By the way, the function could return a Dict[str, Set[Task]]. Am I right?" => It will return a list that doesn't have duplicated tasks. Is it a set -> not necessarily, it's still a list object just without duplication, its function and attributes are still that of a list to keep it consistent with what we have in pipeline.py